### PR TITLE
Render single college query from the selected college list view item

### PR DESCRIPTION
**Pull request by MikolJiron**

### CODE REVIEW

**Review by Lorraine-Lyu — APPROVED** [review]

File: src/main/webapp/js/finscholar/singlepageview.js [hunk 1]
```diff
@@ -29,7 +30,7 @@ class SinglePageView extends BasicView {
   constructor(dataHandler, template) {
     super();
     /**
-     * @private @const {!ScholarshipDataHandler} dataHandler_
+     * @private @const {!ScholarshipDataHandler | !CollegeDataHandler} dataHandler_
```
Comment (Lorraine-Lyu) [inline src/main/webapp/js/finscholar/singlepageview.js]: I'll create a common datahandler today. You'll need to change the type to parent class later.

File: src/main/webapp/js/datahandlers/collegepagedatahandler.js [hunk 2]
```diff
@@ -19,114 +19,52 @@
...
-const ERROR_MESSAGE_JSON_UNDEFINED = 'JSON object undefined.';
-const OCCURRENCE_JSON_UNDEFINED =
-    'A network error has occurred. Failed to load college data.';
-const ACTION_JSON_UNDEFINED =
-    'Please reload the page or select a different college.';
-const OCCURRENCE_SERVER_ERROR =
-    'A database error occurred. Failed to render college data.';
-const ACTION_SERVER_ERROR = 'The database failed to retrieve the college. \
-      This college may not exist. Please reload the page or select a different college.';
+class CollegeDataHandler {
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/collegepagedatahandler.js]: You probably want to wait for Lorraine's refactor and base data handler class to be merged before moving forward with this.

File: src/main/webapp/js/datahandlers/collegepagedatahandler.js [hunk 3]
```diff
@@ -19,114 +19,52 @@
...
-      cumulativeMedianDebt: json['cumulativeMedianDebt']
-    });
-  } else {
-    throw new Error(ERROR_MESSAGE_JSON_UNDEFINED);
+  /** 
+   * @returns {Promise<number>} 
+   * The max number of colleges to render. 
+   */
+  async getTotalNumber() {
+    return 200;
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/collegepagedatahandler.js]: Future work: please make a real API call to get this number.

File: src/main/webapp/js/finscholar/singlepageview.js [hunk 4]
```diff
@@ -29,7 +30,7 @@ class SinglePageView extends BasicView {
   constructor(dataHandler, template) {
     super();
     /**
-     * @private @const {!ScholarshipDataHandler} dataHandler_
+     * @private @const {*} dataHandler_
```
Comment (tinagrasso) [inline src/main/webapp/js/finscholar/singlepageview.js]: Don't use * here. Lorraine create a base class for this.

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Comment by Lorraine-Lyu** [comment]
@googlebot I consent.

File: src/main/webapp/js/datahandlers/scholarshipdatahandler.js [hunk 5]
```diff
@@ -80,11 +80,12 @@ class ScholarshipDataHandler extends SinglePageDataHandler {
   }
 
   /**
+   * @param {string} id 
    * @returns {string} path
    * @private
    */
-  getRequestPath_() {
-    return SCHOLARSHIP_ENDPOINT;
+  getRequestPath_(id) {
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/scholarshipdatahandler.js]: @override?

File: src/main/webapp/js/datahandlers/collegepagedatahandler.js [hunk 6]
```diff
@@ -51,77 +43,56 @@ const ACTION_SERVER_ERROR = 'The database failed to retrieve the college. \
...
-    json = await loadCollegeJson_();
-  } catch (error) {
-    loadErrorPage_(
-        element, OCCURRENCE_JSON_UNDEFINED, ACTION_JSON_UNDEFINED, error);
+  /**
+   * @param {Object} jsonResponse
+   * @returns {Object} Json data needed for rendering page.
+   * @private
+   */
+  getJsonData_(json) {
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/collegepagedatahandler.js:67]: Please add @override to all overridden methods.

File: src/main/webapp/js/datahandlers/singlepagedatahandler.js [hunk 7]
```diff
@@ -45,16 +44,28 @@ class SinglePageDataHandler {
...
    */
-  getRequestPath_() {}
+  getRequestPath_(id) {}
 
     /**
    * This method converts from scholarship JSON object to a JS object map, 
    *  which will be used to render the scholarship page soy template.
    * @param {*} data - The JSON object to be converted.
    * @return {Object} - The object map representing a scholarship's data.
+   * @private
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/singlepagedatahandler.js]: Flip @abstract and @private. The order usually goes params, return, everything else, visibility.

**Review by tinagrasso — APPROVED** [review]
LGTM! Please make changes noted in comments.

**Review by hzimz — APPROVED** [review]

**Review by CarlosM10011 — APPROVED** [review]

**Pull request merged**